src/peerbook/libp2p_peerbook.erl

```diff
@@ -340,32 +335,12 @@ handle_gossip_data(_StreamPid, DecodedList, Handle) ->
     noreply.
 
 -spec init_gossip_data(peerbook()) -> libp2p_group_gossip_handler:init_result().
-init_gossip_data(Handle=#peerbook{tid=TID}) ->
-    [{_, EligiblePeerKeys}] = ets:lookup(TID, peerbook_eligible_gossip_peers),
-    case EligiblePeerKeys of
-        [] ->
-            ok;
-        _ ->
-            SelectPeer = fun F(0) ->
-                                 {error, not_found};
-                             F(Try) ->
-                                 %% Pick a random peer from the set of eligilble peers
-                                 SelectedPeerKey = lists:nth(rand:uniform(length(EligiblePeerKeys)),
-                                                             EligiblePeerKeys),
-                                 case get(Handle, SelectedPeerKey) of
-                                     {ok, P} -> {ok, P};
-                                     {error, _} -> F(Try-1)
-                                 end
-                         end,
-            %% Try 10 times to pick a random peer that is actually
-            %% available in the peerbook
-            case SelectPeer(10) of
-                {ok, Peer} -> {send, libp2p_peer:encode_list([Peer])};
-                {error, not_found} -> ok
-            end
+init_gossip_data(Peerbook) ->
+    case random(Peerbook, [], fun eligible_gossip_peer/1) of
```

Comment:
It'll be interesting to see how this behaves in the field. I.E. it could return non eligible peers a long time which could affect gossip.

Reply:
It'll never return a non-eligible peer, it'll just give up if it doesn't find one after fifteen tries.  I'll increase the number of retries so that it effectively will always find one.